monkey/tests/unit_tests/monkey_island/cc/resources/test_agent_signals.py

```diff
@@ -0,0 +1,70 @@
+from http import HTTPStatus
+from unittest.mock import MagicMock
+from uuid import UUID
+
+import pytest
+from tests.common import StubDIContainer
+
+from monkey_island.cc.models import AgentSignals as Signals
+from monkey_island.cc.repository import RetrievalError, StorageError
+from monkey_island.cc.services import AgentSignalsService
+
+TIMESTAMP_1 = 123456789
+TIMESTAMP_2 = 123546789
+
+SIGNALS_1 = Signals(terminate=TIMESTAMP_1)
+SIGNALS_2 = Signals(terminate=TIMESTAMP_2)
+
+AGENT_ID_1 = UUID("c0dd10b3-e21a-4da9-9d96-a99c19ebd7c5")
+AGENT_ID_2 = UUID("9b4279f6-6ec5-4953-821e-893ddc71a988")
+
+
+@pytest.fixture
+def mock_agent_signals_service():
+    return MagicMock(spec=AgentSignalsService)
+
+
+@pytest.fixture
+def flask_client_builder(build_flask_client, mock_agent_signals_service):
+    def inner(side_effect=None):
+        container = StubDIContainer()
+        container.register_instance(AgentSignalsService, mock_agent_signals_service)
+
+        with build_flask_client(container) as flask_client:
+            return flask_client
+
+    return inner
+
+
+@pytest.fixture
+def flask_client(flask_client_builder):
+    return flask_client_builder()
```

Comment:
If this just returns `flask_client_builder()`, then there's no need for `flask_client_builder`

```suggestion
@pytest.fixture
def flask_client(build_flask_client, mock_agent_signals_service):
    container = StubDIContainer()
    container.register_instance(AgentSignalsService, mock_agent_signals_service)

    with build_flask_client(container) as flask_client:
        yield flask_client
```